test/utils/readRowsImpl.ts

```diff
@@ -431,7 +437,17 @@ export class ReadRowsImpl {
           readRowsRequestHandler.cancelled = true;
           break;
         }
+
+        if (deadlineExceededError) {
+          debugLog('sending deadline exceeded error');
+          const error = new GoogleError('Deadline exceeded');
+          error.code = Status.DEADLINE_EXCEEDED;
+          stream.emit('error', error);
+          readRowsRequestHandler.cancelled = true;
+          break;
+        }
```

Comment:
This looks like it will simulate the right behaviour. You could put this in the `chunkIdx === this.errorAfterChunkNo` block above and specify `errorAfterChunkNo` when defining the service and that might be less code, but this will work.